src/Tinkoff.ISA.Providers.Jira/JiraKnowledgeProvider.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using Atlassian.Jira;
+using Microsoft.Extensions.Options;
+using Tinkoff.ISA.Core;
+using Tinkoff.ISA.Core.Documents;
+using JiraClient = Atlassian.Jira.Jira;
+
+namespace Tinkoff.ISA.Providers.Jira
+{
+    public class JiraKnowledgeProvider : IKnowledgeProvider
+    {
+        private readonly JiraClient _jiraClient;
+        private readonly List<string> _projectNames;
+        private readonly int _batchSize;
+        private readonly string _baseUrl;
+
+        public JiraKnowledgeProvider(JiraClient jiraClient, IOptions<JiraProviderSettings> options)
```

Comment:
I remember that we consider to send this parameters in Startup.cs when we consider which parameters we should set. And in main application engineers can consider how to store these parameters. In this case we create library which is strong coupled to IOptions

Reply:
Do you mean left IOptions in `ServiceCollectionsExtensions`? But does it make sense? If they are located in the same assembly. Moreover, engineers are free to construct new providers as they want, one requirement is to implement a common interface? Am I wrong?

Reply:
When you want to setup Jira provider you just send all the required parameters. And  for that parameters we have class just get all parameters from setup object and use it. And in future engineers can consider how to send these parameters to jira provider.